e2e/internal/kubeclient/deploy.go

```diff
@@ -31,6 +31,8 @@ const (
 	Ready
 	// InitContainersRunning waits until all initial containers of all pods of the resource are running.
 	InitContainersRunning
+	// StartingBlocked waits until a specific FailedCreatePodSandBox Event is detected which indicates that the container does not start.
+	StartingBlocked
```

Comment:
The `WaitCondition` type is documented to be tied to `WaitFor`. On the other hand, `WaitForEvent` does not use any of the other conditions. Should this be a separate enum? Or maybe a predicate `func (e Event) bool`?

Reply:
Changed this to seperate enum!